src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineControl.xaml.cs

```diff
@@ -0,0 +1,330 @@
+            }
+
+            documentSymbolModel.Children = documentSymbolModelChildren;
+            return documentSymbolModel;
+        }
+
+        private void Search(object sender, EventArgs e)
+        {
+            if (SymbolsTreeItemsSource is not null)
+            {
+                if (searchBox.Text == ServicesVSResources.Search_Document_Outline || string.IsNullOrWhiteSpace(searchBox.Text))
+                {
+                    symbolTree.ItemsSource = SymbolsTreeItemsSource;
+                }
+                else
+                {
+                    var documentSymbols = new List<DocumentSymbolViewModel>();
+                    foreach (var item in SymbolsTreeItemsSource)
+                    {
+                        if (DocumentOutlineHelper.SearchNodeTree(item, searchBox.Text))
+                        {
+                            documentSymbols.Add(item);
+                        }
+                    }
+
+                    symbolTree.ItemsSource = documentSymbols;
+                }
+            }
+        }
+
+        private void Sort(SortOption sortOption, FunctionId functionId)
+        {
+            if (SymbolTreeInitialized)
+            {
+                Logger.Log(functionId);
+                // If there is an active search, only sort the filtered nodes
+                if (!(searchBox.Text == ServicesVSResources.Search_Document_Outline || string.IsNullOrWhiteSpace(searchBox.Text)))
+                {
+                    symbolTree.ItemsSource = DocumentOutlineHelper.Sort(
+                        new List<DocumentSymbolViewModel>((IEnumerable<DocumentSymbolViewModel>)symbolTree.ItemsSource), sortOption);
+                }
+                else
+                {
+                    var sortedDocumentSymbolModels = DocumentOutlineHelper.Sort(SymbolsTreeItemsSource, sortOption);
+                    SymbolsTreeItemsSource = sortedDocumentSymbolModels;
+                    symbolTree.ItemsSource = sortedDocumentSymbolModels;
+                }
+            }
+        }
+
+        private void SortByName(object sender, EventArgs e)
+        {
+            Sort(SortOption.Name, FunctionId.DocumentOutline_SortByName);
+        }
+
+        private void SortByOrder(object sender, EventArgs e)
+        {
+            Sort(SortOption.Order, FunctionId.DocumentOutline_SortByOrder);
+        }
+
+        private void SortByType(object sender, EventArgs e)
+        {
+            Sort(SortOption.Type, FunctionId.DocumentOutline_SortByType);
+        }
+
+        // When symbol node clicked, selects corresponding code
+        private void JumpToContent(object sender, EventArgs e)
+        {
+            RoslynDebug.AssertNotNull(Snapshot);
+            if (sender is StackPanel panel && panel.DataContext is DocumentSymbolViewModel symbol)
+            {
+                if (symbol.StartLine >= 0 && symbol.StartLine < Snapshot.LineCount)
```

Comment:
1. generally, we don't store line/column, we store positions.
2. the position will be correct relative to the snapshot when the symbols were created, but that snapshot may have changed.  we should use tracking/mapping points to map the original position to the current snapshot.

Reply:
@CyrusNajmabadi I changed it to store the start/end positions for a document symbol model and used snapshot points to ensure the position is mapped to the current snapshot. Feel free to let me know your thoughts.